src/dotnet-interactive/CommandLine/CommandLineParser.cs

```diff
@@ -250,8 +259,17 @@ Command HttpServer()
                 command.Handler = CommandHandler.Create<StartupOptions, KernelHttpOptions, IConsole, InvocationContext>(
                     (startupOptions, options, console, context) =>
                     {
-                        RegisterKernelInServiceCollection(services, startupOptions, options.DefaultKernel);
-
+                        RegisterKernelInServiceCollection(
+                            services,
+                            startupOptions,
+                            options.DefaultKernel, serviceCollection =>
+                        {
+                            serviceCollection.AddSingleton(_ =>
+                            {
+                                var frontendEnvironment = new BrowserFrontendEnvironment();
```

Comment:
What happens when `FrontEndEnvironment` is resolved?

Reply:
I still need to register for the discovery handler

Reply:
all subtype of FrontendEnvironment are also register with as base type and resolved as base type